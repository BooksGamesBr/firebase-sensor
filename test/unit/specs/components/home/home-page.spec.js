import { shallow, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import HomePage from '@/components/home/home-page'
import { FakeRooms, FakeCategories } from '../../../data/fake-data'

function generateStubs (cmp) {
  return Object.values(cmp.components).reduce((stubs, stubCmp) => {
    const dashName = stubCmp.name
      .replace(/([a-z])([A-Z])/g, '$1-$2')
      .toLowerCase()
    stubs[dashName] = {
      render (createElement) {
        return createElement(dashName, this.$slots.default)
      }
    }
    return stubs
  }, {})
}

const localVue = createLocalVue()

localVue.use(Vuex)

describe('HomePage.vue', () => {
  let getters
  let store

  beforeEach(() => {
    getters = {
      rooms: () => FakeRooms,
      categoriesSorted: () => FakeCategories
    }

    store = new Vuex.Store({
      getters
    })
  })

  it('shows a page', () => {
    const wrapper = shallow(HomePage, {
      store,
      localVue,
      stubs: generateStubs(HomePage)
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
