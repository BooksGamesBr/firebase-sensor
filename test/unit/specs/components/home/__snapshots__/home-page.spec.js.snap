// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage.vue shows a page 1`] = `
<div>
    <recent-data rooms="[object Object]" categories="[object Object],[object Object]"></recent-data>
    <chart-summary rooms="[object Object]" categories="[object Object],[object Object]"></chart-summary>
</div>
`;
